alphaevolve_gs_engine/src/alphaevolve_gs_engine/services/validation/syntactic_validator.py

```
@@ -0,0 +1,207 @@
+            if process.returncode == 0:
+                logger.info(f"Rego policy '{policy_id}' is syntactically valid.")
+                # stdout might contain the parsed AST, not particularly useful here for just validation
+                return True, "Policy is syntactically valid."
+            else:
+                error_message = stderr.strip()
+                logger.warning(f"Rego policy '{policy_id}' is syntactically invalid. Error: {error_message}")
+                return False, f"Syntactic error: {error_message}"
+
+        except FileNotFoundError:
+            logger.error(f"OPA executable not found at '{self.opa_executable_path}'. Cannot validate policy '{policy_id}'.")
+            return False, f"OPA executable not found. Install OPA and ensure it's in PATH or configured."
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)